sympy/matrices/solvers.py

```diff
@@ -372,10 +372,13 @@ def _LUsolve(M, rhs, iszerofunc=_iszero):
     for i in range(n - 1, -1, -1):
         for j in range(i + 1, n):
             scale = A[i, j]
-            b.zip_row_op(i, j, lambda x, y: dps(x - y * scale))
+            b.zip_row_op(i, j, lambda x, y: dps(x - scale * y))
 
         scale = A[i, i]
-        b.row_op(i, lambda x, _: dps(x / scale))
+        if scale.is_commutative:
+            b.row_op(i, lambda x, _: dps(x / scale))
+        else:
+            b.row_op(i, lambda x, _: dps(scale**-1 * x))
```

Comment:
Is the `if` check needed here rather than just using `scale**-1` always?

I think for commutative Expr there is no difference between `x / scale` and `scale**-1 * x` or maybe I am wrong...

Reply:
I believe it should be the same up to performance and complexity (sympy may be constructing a different srepr). I erred on the side of caution here.

Reply:
https://github.com/sympy/sympy/blob/fcf787332a0fe263d458cc24dd0ef558627160ee/sympy/core/expr.py#L287-L294